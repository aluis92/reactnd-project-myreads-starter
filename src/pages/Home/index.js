import React from 'react';
import { Link } from 'react-router-dom';
import Bookshelf from '../../modules/Bookshelf';

function Home ({
    bookList
}) {
    return (
        <div className="list-books">
            <div className="list-books-title">
                <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
                <div>
                    <Bookshelf
                        title="Currently Reading"
                        books={bookList.currentlyReading}
                    />
                    <Bookshelf
                        title="Want To Read"
                        books={bookList.wantToRead}
                    />
                    <Bookshelf
                        title="Read"
                        books={bookList.read}
                    />
                </div>
            </div>
            <div className="open-search">
                <Link to="/search">Add a book</Link>
            </div>
        </div>
    );
}

export default Home;
