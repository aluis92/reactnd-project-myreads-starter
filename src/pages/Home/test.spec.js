import React from 'react';
import {
    shallow,
} from 'enzyme';

import Home from './index';

describe('Home component', () => {
    const defaultHomeProps = {
        updateBook: jest.fn(),
    };

    it('should render without crashing', () => {
        expect(shallow(
            <Home {...defaultHomeProps} />,
        )).toMatchSnapshot();
    });

    it('should render shelves if there is one or more books inside bookShelf', () => {
        const homeProps = {
            bookShelves: {
                read: [{}, {}],
                wantToRead: [{}],
                anotherShelf: [],
            },
            updateBook: jest.fn(),
        };

        const home = shallow(
            <Home {...homeProps} />,
        );

        expect(home.find('Bookshelf').length).toBe(2);
    });
});
