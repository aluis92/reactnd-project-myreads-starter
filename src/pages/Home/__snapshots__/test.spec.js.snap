// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render without crashing 1`] = `
<ListBooks
  title="MyReads"
>
  <div />
</ListBooks>
`;
